source/allocator.c

```diff
@@ -51,9 +51,13 @@ static void *s_default_malloc(struct aws_allocator *allocator, size_t size) {
     const size_t alignment = sizeof(void *) * (size > PAGE_SIZE ? 8 : 2);
 #if !defined(_WIN32)
     void *result = NULL;
-    return (posix_memalign(&result, alignment, size)) ? NULL : result;
+    posix_memalign(&result, alignment, size);
+    AWS_PANIC_OOM(result, "posix_memalign failed to allocate memory");
+    return result;
 #else
-    return _aligned_malloc(size, alignment);
+    void *mem = _aligned_malloc(size, alignment);
+    AWS_FATAL_POSTCONDITION(mem && "_aligned_malloc failed to allocate memory");
```

Comment:
The explanation should be after a comma.
```c
AWS_FATAL_POSTCONDITION(mem != NULL, "_aligned_malloc failed to allocate memory");
```

@JonathanHenson

Reply:
we changed it to a different macro, see latest diff.

Reply:
Am I looking at the wrong diff? I see a new macro at `assert.h` called `AWS_PANIC_OOM`, but I don't see any updates to `AWS_FATAL_POSTCONDITION`. So if you we don't add a comma here, we will still treat the explanation as part of the condition.

Reply:
```suggestion
    AWS_FATAL_POSTCONDITION(mem, "_aligned_malloc failed to allocate memory");
```